### PR TITLE
[Docs] RayDP Documentation

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This PR introduces some example documentation for how to use RayDP and points to the RayDP docs page.

Here's an image of what the documentation looks like. 

image

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @ carsonwang
 
Can you review it?

File: doc/source/raydp.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+The RayDP project offers PySpark, using Ray as Spark's cluster manager. It also
+offers convenient tooling to integrate PySpark's
+`DataFrame API <https://spark.apache.org/docs/latest/api/python/pyspark.sql.html>`_
+with your favorite deep learning libarires (PyTorch and TensorFlow).
+
+=======================
+Installing RayDP
+=======================
+
+RayDP can be installed from PyPI and supports PySpark 3.0 and 3.1.
```
Comment (rkooo567) [inline doc/source/raydp.rst:14]: Mention ray >= 1.2.0

File: doc/source/raydp.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  )
+
+  estimator.fit_on_spark(df, test_df=None)
+
+  pytorch_model = estimator.get_model()
+
+
+==========
+Conclusion
+==========
```
Comment (rkooo567) [inline doc/source/raydp.rst:103]: Remove conclusion

File: doc/source/raydp.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+RayDP combines your Spark and Ray clusters, making it easy to do large scale
```
Comment (rkooo567) [inline doc/source/raydp.rst:105]: Let's push to intro.

**Review by rkooo567 — APPROVED** [review]

File: doc/source/raydp.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
+*********************
+RayDP (Spark on Ray)
+*********************
```
Comment (rkooo567) [inline doc/source/raydp.rst:3]: Make sure the legnth of line and characters are the same

**Comment by rkooo567** [comment]
cc @carsonwang

**Comment by rkooo567** [comment]
I will merge the PR first, but please let us know if you'd like to add more contents here. (and feel free to create correction PR)!

**Comment by carsonwang** [comment]
Thanks for adding this! Sorry for delayed response due to Chinese new year vacation.

**Pull request merged**